src/styled.js

```diff
@@ -59,7 +59,7 @@ function styled(tag, forwardRef) {
             }
 
             // Handle the forward props filter if defined and _as is a string
-            if (fwdProp && _as[0]) {
+            if (fwdProp && _as) {
```

Comment:
Hey @kitten, so this is used in here to flag that only at the DOM tag level we need to run the fwdProp function in order to remove them from appearing onto the element as attributes. So this essentially is a poor man version of `typeof _as === "string"`.
```suggestion
            if (fwdProp && _as[0]) {
```